device/cuda/CMakeLists.txt

```diff
@@ -44,10 +44,18 @@ traccc_add_library( traccc_cuda cuda TYPE SHARED
   # Fitting
   "include/traccc/cuda/fitting/fitting_algorithm.hpp"
   "src/fitting/fitting_algorithm.cu")
-  
+if(TRACCC_SETUP_PROFILER)
 target_compile_options( traccc_cuda
-  PRIVATE $<$<COMPILE_LANGUAGE:CUDA>:--expt-relaxed-constexpr> )
+  PRIVATE $<$<COMPILE_LANGUAGE:CUDA>:--expt-relaxed-constexpr -Xcompiler -finstrument-functions -fPIC > )
 target_link_libraries( traccc_cuda
-  PUBLIC traccc::core detray::core detray::utils vecmem::core covfie::core
-  PRIVATE CUDA::cudart traccc::device_common vecmem::cuda )
+        PUBLIC traccc::core detray::core detray::utils vecmem::core covfie::core
+        PRIVATE CUDA::cudart traccc::device_common vecmem::cuda -lnvToolsExt)
+else()
+    target_compile_options( traccc_cuda
+            PRIVATE $<$<COMPILE_LANGUAGE:CUDA>:--expt-relaxed-constexpr>)
+    target_link_libraries( traccc_cuda
+            PUBLIC traccc::core detray::core detray::utils vecmem::core covfie::core
+            PRIVATE CUDA::cudart traccc::device_common vecmem::cuda)
+endif(TRACCC_SETUP_PROFILER)
```

Comment:
This is not the best way to solve this. In an unusual turn of events CMake actually has halfway decent syntax for this, which they refer to as "generator expressions". You should write something like `$<$<TRACCC_SETUP_PROFILER>:-Xcompiler -finstrument-functions -fPIC>`. Or, as we might prefer to generate position-independent code in any case: `-Xcompiler -fPIC  $<$<TRACCC_SETUP_PROFILER>:-finstrument-functions>`. To be clear, the use of this is that we would not need to duplicate this much code between the true and false branches of the if-statement, as we could do without the if entirely.

More documentation on generator expressions is available at https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html.

Reply:
Implemented the generator expression using the following line:

`$<$<BOOL:${TRACCC_SETUP_PROFILER}>:-Xcompiler -finstrument-functions -fPIC>>`